File: 1. Two Sum.cpp

#include<iostream>
#include<unordered_map>
#include<vector>
using namespace std;
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;
        for(int i=0; i<nums.size();i++){
            int needs = target- nums[i];
            if(mp.count(needs)){
                return {mp[needs],i};
            }
            mp[nums[i]] =i;
            
        }
        return {};
    }
};

File: 1. Two Sum.py
class Solution:
    def twoSum(self, nums, target):
        mp = {}
        
        for i in range(len(nums)):
            needs = target - nums[i]
            
            if needs in mp:
                return [mp[needs], i]
            
            mp[nums[i]] = i
        
        return []